ocaml/sdk-gen/csharp/gen_csharp_binding.ml

```diff
@@ -867,68 +858,45 @@ and has_unknown_entry contents =
 
 (* ------------------- category: maps *)
 and gen_maps () =
-  let out_chan = open_out (Filename.concat destdir "Maps.cs") in
-  finally (fun () -> gen_maps' out_chan) ~always:(fun () -> close_out out_chan)
-
-and gen_maps' out_chan =
-  let print format = fprintf out_chan format in
-
-  print
-    "%s\n\n\
-     using System;\n\
-     using System.Collections;\n\
-     using System.Collections.Generic;\n\n\
-    \     namespace XenAPI\n\
-     {\n\
-    \    internal class Maps\n\
-    \    {" Licence.bsd_two_clause ;
-
-  TypeSet.iter (gen_map_conversion out_chan) !maps ;
-
-  print "\n    }\n}\n"
-
-and gen_map_conversion out_chan = function
-  | Map (l, r) ->
-      let print format = fprintf out_chan format in
-      let el = exposed_type l in
-      let el_literal = exposed_type_as_literal l in
-      let er = exposed_type r in
-      let er_literal = exposed_type_as_literal r in
-
-      print
-        "\n\
-        \        internal static Dictionary<%s, %s> \
-         ToDictionary_%s_%s(Hashtable table)\n\
-        \        {\n\
-        \            Dictionary<%s, %s> result = new Dictionary<%s, %s>();\n\
-        \            if (table != null)\n\
-        \            {\n\
-        \                foreach (string key in table.Keys)\n\
-        \                {\n\
-        \                    try\n\
-        \                    {\n\
-        \                        %s k = %s;\n\
-        \                        %s v = %s;\n\
-        \                        result[k] = v;\n\
-        \                    }\n\
-        \                    catch\n\
-        \                    {\n\
-        \                       // continue\n\
-        \                    }\n\
-        \                }\n\
-        \            }\n\
-        \            return result;\n\
-        \        }\n\n"
-        el er
-        (sanitise_function_name el_literal)
-        (sanitise_function_name er_literal)
-        el er el er el
-        (simple_convert_from_proxy "key" l)
-        er
-        (convert_from_proxy_hashtable_value "table[key]" r)
-  (***)
-  | _ ->
-      assert false
+  let mapList = List.rev (TypeSet.fold (fun x acc -> x :: acc) !maps []) in
+  let json =
+    `O
+      [
+        ( "all_maps"
+        , `A
+            (List.map
+               (fun x ->
+                 match x with
```

Comment:
You can skip some indentation with
```suggestion
               (function
```

Reply:
It saves one line, so happy to remove it, but the formatter doesn't seem to change the identation.